server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -97,11 +93,33 @@ private static int dotCount(String path) {
      * Returns the mapped field type for the given field name.
      */
     MappedFieldType get(String field) {
-        MappedFieldType fieldType = fullNameToFieldType.get(field);
-        if (fieldType != null) {
-            return fieldType;
+        GuardedFieldTypeLookup guardedLookup = new GuardedFieldTypeLookup();
+        return guardedLookup.get(field);
+    }
+
+    private class GuardedFieldTypeLookup {
```

Comment:
I wonder if it's really valuable to support aliases to other aliases. The same goes for aliases to other runtime fields. I can't see a strong use case for this -- the user can already easily rename fields in the `runtime` section, or just add the definition twice if they need both fields present. Aliases are most helpful because you can't update a 'standard' field without reindexing.

Do you have some use cases in mind related to runtime fields? I'm curious because supporting this adds complexity here and in `SearchExecutionContext` to prevent loops. I wonder if it'd be the right trade-off to require the `path` to point to a standard field (defined under `properties`).